subset_vds.py

```diff
@@ -0,0 +1,373 @@
+    MatrixTable,
+    Table,
+    export_plink,
+    export_vcf,
+    get_reference,
+    parse_locus_interval,
+    split_multi_hts,
+)
+from hail.vds import filter_intervals, filter_samples, read_vds, to_dense_mt
+from hail.vds.variant_dataset import VariantDataset
+
+
+def check_output_already_exists(output_format: list[str], infile_name: str) -> None:
+    """Check for existence of output files
+
+    Args:
+        output_format (list[str]): list of output formats to generate files for
+        infile_name (str): base name of the dataset
+
+    Returns:
+        Tuple[bool, str]: A tuple to indicate if any output files already exist, and which ones they are.
+    """
+    output_errors: str = ""
+    files_exist_errors: bool = False
+    for format in output_format:
+        if format == "vds" and to_path(output_path(f"{infile_name}_subset")).exists():
+            output_errors += f"The output VDS {infile_name}_subset already exists. Refusing to overwrite it.\n"
+            files_exist_errors = True
+        if (
+            format == "bed"
+            and to_path(output_path(f"{infile_name}_subset.bed")).exists()
+        ):
+            output_errors += f"The output {to_path(output_path(infile_name))}_subset.bed fileset exists. Refusing to overwrite it.\n"
+            files_exist_errors = True
+        if (
+            format == "vcf"
+            and to_path(output_path(f"{infile_name}_subset.vcf.bgz")).exists()
+        ):
+            output_errors += f"The output file {to_path(output_path(infile_name))}_subset.vcf.bgz exists. Refusing to overwrite it.\n"
+            files_exist_errors = True
+    if files_exist_errors:
+        raise FileExistsError(f"{output_errors}")
+
+
+def parse_intervals(intervals: str) -> list[str]:
+    """read intervals from either the command line or a file
+
+    Args:
+        intervals (str): a string of either comma separated intervals, or a path to a file of intervals (one per line) in gcs
+
+    Raises:
+        ValueError: the intervals need to match the following pattern: ^[0-9cxymt]+ (does it start with some kind of contig identifier?)
+
+    Returns:
+        list[str]: a list of strings representing intervals
+    """
+    interval_list: list[str] = []
+    problem_intervals: list[str] = []
+    if to_path(intervals).exists():
+        with open(to_path(intervals), "rt") as interval_file:
+            interval_list = interval_file.readlines()
+    else:
+        interval_list = intervals.split(",")
+    for interval in interval_list:
+        if not fullmatch("^[0-9chrxymt]+", interval.split(":")[0].lower()):
+            problem_intervals.append(interval)
+    if len(problem_intervals) > 0:
+        raise ValueError(
+            f"The list of supplied intervals contains the following incorrectly formatted contigs: {[problem for problem in problem_intervals]}"
+        )
+    return interval_list
+
+
+def convert_intervals_to_locus(interval_list: list[str]) -> list[IntervalExpression]:
```

Comment:
I think a lot of this code block/validation you would get for free with just calling `parse_locus_interval` - it tolerates `chr`, `chr:start-stop`, `chr1-chr2`, and would give you positional validation based on the reference.

If this was partially taken from [here](https://github.com/populationgenomics/analysis-runner/blob/main/scripts/subset_matrix_table.py#L150), then yes, I probably also wasted some time.

One piece of validation you might be missing here is checking for a `chr` prefix on the chromosome name, or a check that the chromosome name appears in the hail.reference.contigs

Reply:
I did use that as a reference, yes. Not familiar with using hail locuses yet.
If I can delete all this checking code and basically rely on `parse_locus_interval` to do all the checking for me, then I'm happy to do that. Thoughts?

Reply:
It looks like you've already moved in that direction. Sounds good to me - if you just force a user to provide one of the Hail supported input formats, and if there are any invalid ones print a link to the Hail docs on which ones are valid, then it's still pretty flexible, and takes the burden off you writing the validation